xprof/xprof.rb.in

```diff
@@ -370,145 +420,166 @@ def enable_events_cl(lttng_session_uuid, channel_name, tracing_mode: 'default',
-  ENV['LTTNG_HOME'] = lttng_home
+  ENV['LTTNG_HOME'] = lttng_home_dir
   # Only local master spawn LTTNG daemon and start session
-  setup_lttng(lttng_output, lttng_session_uuid, lttng_home, backends) if mpi_local_master?
+  setup_lttng(backends) if mpi_local_master?
   local_barier('waiting_for_lttng_setup')
   # Launch User Command
   launch_usr_bin(h, usr_argv)
 
   # We need to be sure that all the local ranks finished
   # before local master stops the lttng session
   local_barier('waiting_for_application_ending')
-  teardown_lttng(lttng_session_uuid, lttng_home) if mpi_local_master?
-
-  # Ensure that all traces have been written so we can post-process them
-  global_barrier(global_barrier_h) if mpi_local_master?
-
-  # Master will move all the folders to a better folder
-  rename_to_human_readable_folder(File.dirname(lttng_output)) if mpi_master?
-end
-
-#    _                                   _
-#   |_)  _. |_   _  | _|_ ._ _.  _  _     )
-#   |_) (_| |_) (/_ |  |_ | (_| (_ (/_   /_
-#
+  return unless mpi_local_master?
 
-def last_trace_saved
-  Dir[File.join(env_fetch_first('HOME'), 'lttng-traces', 'thapi--*')].max_by { |f| File.mtime(f) }
+  teardown_lttng
+  # Preprocess trace
+  on_node_processing(backends, global_barrier_h)
 end
 
-# TODO: Use babeltrace_thapi as a LIB not a binary
-def postprocess(folder, _options)
-  return unless mpi_master?
+def global_processing(folder)
+  raise unless mpi_master?
 
   LOGGER.info { "postprocess #{folder}" }
-  puts("Trace Location #{folder}")
   backends = OPTIONS[:'backend-name'].join(',')
   backend_level = OPTIONS[:backend].filter { |nl| nl.include?(':') }.join(',')
 
   if OPTIONS.include?(:trace)
-    opts = 'trace'
-    opts << '--restrict '
-    opts << '--context '
-    opts << "--backend #{backends} "
+    if folder.include?('_interval') || folder.include?('_aggreg')
+      raise 'Cannot show the trace of an interval or aggregated trace'
+    end
+
+    opts = ['trace']
+    opts << '--restrict'
+    opts << '--context'
+    opts << "--backend #{backends}"
   elsif OPTIONS.include?(:timeline)
-    opts = 'timeline'
-    opts << "--backend #{backends} "
+    raise 'Cannot show the trace of an aggreg trace' if folder.include?('_aggreg')
+
+    opts = ['timeline']
+    opts << "--backend #{backends}"
   else
-    opts = 'tally '
-    opts << '--display_kernel_verbose true ' if OPTIONS.include?(:'kernel-verbose')
-    opts << '--display_metadata true ' if OPTIONS.include?(:metadata)
-    opts << "--display_name_max_size #{OPTIONS[:'max-name-size']} "
-    opts << '--display_mode json ' if OPTIONS.include?(:json)
-    opts << "--backend #{backends} "
+    opts = ['tally']
+    opts << '--display_kernel_verbose true' if OPTIONS.include?(:'kernel-verbose')
+    opts << '--display_metadata true' if OPTIONS.include?(:metadata)
+    opts << "--display_name_max_size #{OPTIONS[:'max-name-size']}"
+    opts << '--display_mode json' if OPTIONS.include?(:json)
```

Comment:
```suggestion
    opts << '--display_mode' << 'json' if OPTIONS.include?(:json)
```
Wouldn't that prevent issues like space escaping if we end up using the Array in an exec call?

Reply:
So weird issue with `-` and `_` in the variable. I got some error `incorrect --display-name-max-size did you mean display_name_max_size`

Reply:
If we do it for `json` do you want to do it for all the other? like `max-name-size`  `true` and other?

Reply:
Yes it was that
```
    opts = ['tally']
    opts << '--display_kernel_verbose' if OPTIONS.include?(:'kernel-verbose')
    opts << '--display_metadata' if OPTIONS.include?(:metadata)
    opts << "--display_name_max_size" << OPTIONS[:'max-name-size'].to_s
    opts << '--display_mode' << 'json' if OPTIONS.include?(:json)
    opts << "--backend" << backends
    opts << "--backend_level" << backend_level unless backend_level.empty?
    opts << '--display' << 'extended' if OPTIONS.include?(:extended)
  end

  opts << '--no-muxer' if folder.include?('thapi_interval') || folder.include?('thapi_aggreg')

  cmd = ["#{BINDIR}/babeltrace_thapi"] + opts + ["--", folder]
  ```

Is working